bakery/src/tests/test_bakery_scripts.py

```diff
@@ -857,3 +859,245 @@ def test_gdocify_book(tmp_path, mocker):
         "//x:a[@href]", namespaces={"x": "http://www.w3.org/1999/xhtml"},
     ):
         assert expected_links_by_id[node.attrib["id"]] == node.attrib["href"]
+
+
+def test_copy_resource_s3(tmp_path, mocker):
+    """Test copy_resource_s3 script"""
+    resource_name = "fffe62254ef635871589a848b65db441318171eb"
+
+    test_resource = os.path.join(TEST_DATA_DIR, resource_name)
+
+    book_dir = tmp_path / "col11762"
+    book_dir.mkdir()
+
+    resources_dir = book_dir / "resources"
+    resources_dir.mkdir()
+    resource = resources_dir / "fffe62254ef635871589a848b65db441318171eb.json"
+
+    test_data_one = open(test_resource, "rb").read()
+    resource.write_bytes(test_data_one)
+
+    dist_bucket = 'distribution-bucket-1234'
+    dist_bucket_prefix = 'master/resources'
+    key = dist_bucket_prefix + '/' + resource_name
+
+    s3_client = boto3.client('s3')
+    s3_stubber = botocore.stub.Stubber(s3_client)
+
+    s3_stubber.add_response(
+        'list_objects',
+        {},
+        expected_params={
+            'Bucket': dist_bucket,
+            'Prefix': dist_bucket_prefix + '/',
+            'Delimiter': '/'
+        }
+    )
+
+    s3_stubber.add_response(
+        "put_object",
+        {},
+        expected_params={
+            'Body': botocore.stub.ANY,
+            'Bucket': dist_bucket,
+            'ContentType': 'application/json',
+            'Key': 'master/resources/fffe62254ef635871589a848b65db441318171eb-unused.json',
+        }
+    )
+
+    s3_stubber.add_response(
+        "put_object",
+        {},
+        expected_params={
+            'Body':  botocore.stub.ANY,
+            'Bucket': dist_bucket,
+            'ContentType': 'image/jpeg',
+            'Key': key,
+        }
+    )
+
+    s3_stubber.activate()
+
+    from unittest.mock import MagicMock
+    mocked_session = boto3.session.Session
+    mocked_session.client = MagicMock(return_value=s3_client)
```

Comment:
Minor: I believe you can avoid the inline import above by using `mocker.MagicMock` via the pytest fixture here.